implementation.js

```diff
@@ -138,6 +152,9 @@ module.exports = function from(arrayLike) {
 	}
 	var items = ES.ToObject(arrayLike);
 
+	// variables for rebuilding array from iterator or string.
+	items = parseES6(items);
+
 	var mapFn, T;
 	if (typeof arguments[1] !== 'undefined') {
 		mapFn = arguments[1];
```

Comment:
I am a bit concerned with observable differences here - the spec mandates the order of Get calls and which exceptions can be thrown when. I'll have to sit and read the steps to be sure, but I think line 162 and 170 need to run before this logic.